sunspec/core/util.py

```diff
@@ -95,6 +96,12 @@ def data_to_double(data):
         return d[0]
 
 def data_to_str(data):
+
+    # Change the data from bytes string to regular string for python 3
+    # compatibility
+    if sys.version_info > (3,):
+        data = str(data, 'latin-1')
```

Comment:
You shouldn't arbitrarily flip between byte sequences and character sequences. Despite the very common lack of understanding of that in py2 code and people often using byte sequences with ASCII as if they are character sequences.  In both py2 and 3 they are unique.

I didn't say that the spec should specify an encoding. The thing is that the library should agree with the spec. That is, either they both have a defined encoding for strings or neither does.

Maybe take a look at http://bit.ly/unipain.